Library/Homebrew/livecheck.rb

```diff
@@ -40,6 +41,16 @@ def skip?
     @skip
   end
 
+  # Sets the strategy instance variable to the argument given or returns the
+  # strategy instance variable when no argument is given.
+  # @param symbol [Symbol] corresponding symbol for desired LivecheckStrategy
```

Comment:
How would I find out what these are?

Reply:
Right now the only way you would know is if you happened to look at [`LivecheckStrategy#strategies`](https://github.com/Homebrew/homebrew-livecheck/blob/master/livecheck/livecheck_strategy.rb#L7-L12) (where the symbol/strategy mapping is done):

```ruby
@strategies = {}
LivecheckStrategy.constants.sort.each do |strategy_symbol|
  key = strategy_symbol.to_s.underscore.to_sym
  strategy = LivecheckStrategy.const_get(strategy_symbol)
  @strategies[key] = strategy
end
@strategies
```

In simple terms, the symbol for a strategy is the name run through `#underscore` (which also makes it lowercase). For the `PageMatch` strategy, the constant is `:PageMatch` which becomes `:page_match`. The only reason we're not using the original symbol (e.g., `:PageMatch`) is because I wasn't sure if something like `strategy :PageMatch` would be appropriate compared to `strategy :page_match`.

However, there's something to be said for using the original symbol (e.g., `:PageMatch`) as it would be very simple to say "the symbol is the name of the strategy but as a symbol (e.g., `:PageMatch`)". The available strategies for a URL are listed as part of livecheck's debug output, so it would also be more comprehensible to see `Strategies:       Sourceforge, PageMatch` and know you can use `strategy :PageMatch` in the `livecheck` block for that URL.

What do you think about this?

---

It's probably also worth noting that we don't hardcode an array of the symbols in `LivecheckStrategy#strategies` because we allow taps to create their own strategies and those need to be included as well. The only tap that did this previously was Linuxbrew/homebrew-xorg but we ended up adding the `Xorg` strategy to livecheck itself (since some X.org formulae are being incorporated into homebrew-core).

Reply:
> Right now the only way you would know is if you happened to look at [`LivecheckStrategy#strategies`](https://github.com/Homebrew/homebrew-livecheck/blob/master/livecheck/livecheck_strategy.rb#L7-L12) (where the symbol/strategy mapping is done):

Would be good to explicitly document how one would do that.

As for the rest of your comment: that's a great explanation and would be good to work into the documentation here!

Reply:
What do you think about using the original symbols from `LivecheckStrategy.constants` (e.g., `:PageMatch`) instead of the `underscore` ones (e.g., `:page_match`)?

There's currently only one formula with a `livecheck` block using `strategy`, so it would be easy to change this now.

Reply:
`:page_match` feels a bit nicer it'd just be good to have one place I could look at to get these references. Maybe that's just the filenames (and we could test they are consistent)?